apps/omg_utils/mix.exs

```diff
@@ -25,5 +25,12 @@ defmodule Utils.MixProject do
   defp elixirc_paths(:prod), do: ["lib"]
   defp elixirc_paths(_), do: ["lib", "test/support"]
 
-  defp deps, do: [{:appsignal, "~> 1.9"}]
+  defp deps do
+    [
+      {:appsignal, "~> 1.9"},
+      {:spandex, "~> 2.4"},
+      {:spandex_datadog, "~> 0.4"},
+      {:decorator, "~> 1.2"}
```

Comment:
`decorator` is included as part of `appsignal`, but adding explicitly here as well for when we remove `appsignal` as a dependency.

Reply:
hm, looks like you've already added the `deps` into the "top-level" `mix.exs`. I think it's enough and appropriate if the dependencies are used across the entire umbrella app. Did it not work if the `deps` were missing in `omg_utils/mix.exs`?

Reply:
When I removed `spandex` from this `mix.exs`, the dialyzer complained (see the circleci build [here](https://circleci.com/gh/omisego/elixir-omg/11307))

```
:0:unknown_function Function Spandex.continue_trace/3 does not exist.
:0:unknown_function Function Spandex.continue_trace_from_span/3 does not exist.
:0:unknown_function Function Spandex.current_context/1 does not exist.
:0:unknown_function Function Spandex.current_span/1 does not exist.
:0:unknown_function Function Spandex.current_span_id/1 does not exist.
:0:unknown_function Function Spandex.current_trace_id/1 does not exist.
:0:unknown_function Function Spandex.distributed_context/2 does not exist.
:0:unknown_function Function Spandex.finish_span/1 does not exist.
:0:unknown_function Function Spandex.finish_trace/1 does not exist.
:0:unknown_function Function Spandex.inject_context/3 does not exist.
:0:unknown_function Function Spandex.span_error/3 does not exist.
:0:unknown_function Function Spandex.start_span/2 does not exist.
:0:unknown_function Function Spandex.start_trace/2 does not exist.
:0:unknown_function Function Spandex.update_span/1 does not exist.
:0:unknown_function Function Spandex.update_top_span/1 does not exist.
:0:unknown_type Unknown type: Spandex.Tracer.opts/0.
```

It seemed like maybe spandex wasn't being included? When I re-added the deps here it fixed the dialyzer. It also appeared that we included `appsignal` in almost every single `mix.exs`, so I thought maybe it was normal to include something more than once.

Reply:
hm, when you mention that I now vaguely recall that there was something strange that caused us to include `appsignal` everywhere too, possibly dialyzer.

Normally it should suffice to include on the top level, but some deps have their quirks...